### PR TITLE
BPM package: allow bonded atoms to be excluded from neighbor lists if BPM cannot break + doc update

### PR DESCRIPTION
**Pull request by jibril-b-coulibaly**
**Summary**

1. Make error message consistent with new keyword values yes/no from #3807 
2. Allow special_bonds coul 0 1 1 when `break no`. There is no concern regarding neighbor list builds happening between bond breakage when there is no bond breakage. Eliminating bonded atoms from neighbor list can be useful for post-processing or to get per BPM pair quantities such as number of contacts.

**Related Issue(s)**

N/A

**Author(s)**

Jibril B. Coulibaly, Sandia National Laboratories (jibril.coulibaly@gmail.com)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Yes

**Implementation Notes**

N/A

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

N/A

### CODE REVIEW

File: doc/src/Howto_bpm.rst [hunk 1]
```diff
@@ -107,7 +107,11 @@ bond lists is expensive.  By setting the lj weight for 1-2 bonds to
 zero, this turns off pairwise interactions.  Even though there are no
 charges in BPM models, setting a nonzero coul weight for 1-2 bonds
 ensures all bonded neighbors are still included in the neighbor list
-in case bonds break between neighbor list builds.
+in case bonds break between neighbor list builds. If bond breakage is
+disabled during a simulation run by setting the *break* keyword to *no*,
+a zero coul weight for 1-2 bonds can be used to exclude bonded atoms
```
Comment (sjplimp) [inline doc/src/Howto_bpm.rst:112]: What about the LJ weights ?  What should they be set to ?

**Review by jtclemm — APPROVED** [review]
This looks like a valid use for LJ[1] = Coul[1] = 0.0 and the PR looks good to me. In the future, it might be good to add a table listing the possible combinations of options, the required LJ/Coul weights, and what they do. We'll have to first think whether any other combinations might be useful for users.

File: doc/src/Howto_bpm.rst (hunk 1 above)
Comment (jtclemm) [reply doc/src/Howto_bpm.rst:112]: From the code, the LJ weights still have to be either 0 1 1 or 1 1 1. Is this right Jibril?

I can't think of a use case for any other option...

Comment (akohlmey) [reply doc/src/Howto_bpm.rst:112]: Please note that the `special_bonds` command is a bit "special". If you specify only the `coul` keyword this will *reset* the `lj` settings. Thus it is recommended to always specify both or use `lj/coul`  (especially in examples or the documentation) so there is no misunderstanding what the settings should be.

Comment (jibril-b-coulibaly) [reply doc/src/Howto_bpm.rst:112]: - If `overlay/pair yes` is used, the lj weights **must be** 1 X Y, regardless of `break yes` or `break no`.

The code currently only checks for LJ[1]==1. It does not enforce X=1 or Y=1, nor coul weights to be 1 1 1.
The main downside of not enforcing LJ[2]=LJ[3]=Coul[1]=Coul[2]=Coul[3]=1 would be the cost of building the 1-3 and 1-4 special bonds if a user selects a value <1. We could enforce that, or leave it as is, please let me know your thoughts @jtclemm @sjplimp 

- If `overlay/pair no` is used, the lj weights **must be** 0 1 1, regardless of `break yes` or `break no`

Let me rephrase the doc to explicitly include `special_bonds lj 0 1 1 coul 0 1 1` per @akohlmey's comment about `coul` resetting `lj`.

Comment (jtclemm) [reply doc/src/Howto_bpm.rst:112]: > We could enforce that, or leave it as is, please let me know your thoughts @jtclemm @sjplimp

Oh right, no limits were placed on the `overlay yes` option. I don't have strong opinions, but I would lean on the side of simply always requiring the LJ[2] = LJ[3] = Coul[2] = Coul[3] = 1.0. The BPM package will already error if used with 3- or 4-body interactions, so this shouldn't add any practical restrictions.

I do have a version of 3-body BPM interactions which I may eventually update and add to the package (and could extend it to 4-body), but I think it's best to worry about those cases later.

Comment (sjplimp) [reply doc/src/Howto_bpm.rst:112]: I think @jtclemm 's comment is the right approach.  Have the user follow the simple rule now - if it needs to be relaxed for a later extension, do that later.

Comment (jibril-b-coulibaly) [reply doc/src/Howto_bpm.rst:112]: Sounds good to me. I updated the error check to enforce LJ[1] = LJ[2] = LJ[3] = Coul[1] = Coul[2] = Coul[3] = 1.0 when `overlay/pair yes` is used and updated the documentation accordingly.

Comment (jtclemm) [reply doc/src/Howto_bpm.rst:112]: Thanks for adding those checks Jibril. I think the PR should be good to go.

**Review by sjplimp — APPROVED** [review]
